impeller/aiks/color_source.h

```diff
@@ -25,6 +25,90 @@ namespace impeller {
 
 struct Paint;
 
+struct LinearGradientData {
+  Point start_point;
+  Point end_point;
+  std::vector<Color> colors;
+  std::vector<Scalar> stops;
+  Entity::TileMode tile_mode;
+  Matrix effect_transform;
+};
+
+struct RadialGradientData {
+  Point center;
+  Scalar radius;
+  std::vector<Color> colors;
+  std::vector<Scalar> stops;
+  Entity::TileMode tile_mode;
+  Matrix effect_transform;
+};
+
+struct ConicalGradientData {
+  Point center;
+  Scalar radius;
+  std::vector<Color> colors;
+  std::vector<Scalar> stops;
+  Point focus_center;
+  Scalar focus_radius;
+  Entity::TileMode tile_mode;
+  Matrix effect_transform;
+};
+
+struct SweepGradientData {
+  Point center;
+  Degrees start_angle;
+  Degrees end_angle;
+  std::vector<Color> colors;
+  std::vector<Scalar> stops;
+  Entity::TileMode tile_mode;
+  Matrix effect_transform;
+};
+
+struct ImageData {
+  std::shared_ptr<Texture> texture;
+  Entity::TileMode x_tile_mode;
+  Entity::TileMode y_tile_mode;
+  SamplerDescriptor sampler_descriptor;
+  Matrix effect_transform;
+};
+
+struct RuntimeEffectData {
+  std::shared_ptr<RuntimeStage> runtime_stage;
+  std::shared_ptr<std::vector<uint8_t>> uniform_data;
+  std::vector<RuntimeEffectContents::TextureInput> texture_inputs;
+};
+
+#if IMPELLER_ENABLE_3D
+struct SceneData {};
+#endif  // IMPELLER_ENABLE_3D
+
+using ColorSourceData = std::variant<LinearGradientData,
+                                     RadialGradientData,
+                                     ConicalGradientData,
+                                     SweepGradientData,
+                                     ImageData,
+                                     RuntimeEffectData,
+                                     std::monostate>;
+
+struct ColorSourceDataVisitor {
```

Comment:
```suggestion
struct CreateContentsVisitor {
```
Since that's what it does.

Reply:
Done